col/src/main/java/vct/col/ast/Rewriter.scala

```diff
@@ -1,8 +1,17 @@
 package vct.col.ast
 
+import vct.col.ast.RewriteHelpers._
+
 class Rewriter extends AbstractRewriter {
   override def dispatch(stat: Statement): Statement = rewriteDefault(stat)
-  override def dispatch(e: Expr): Expr = rewriteDefault(e)
+  override def dispatch(e: Expr): Expr = {
+    e match {
+      case node: Constant.BooleanValue =>
+        new Constant.BooleanValue(node.value)(node.o)
```

Comment:
This was added because the automatically generated class does not generate methods for BooleanValue.

Reply:
That sounds like a bug in the generation code. Can ensure that the methods are generated, or is there something preventing that?

Reply:
it's definitely a bug, I intend to have every node auto-generated, just haven't looked at it yet.

Reply:
Ah, is this because IntegerValue and BooleanValue are nested with the Constant object, instead of top-level case classes?

Reply:
Good guess! I added support for arbitrary nesting of `object`'s though, so potentially not, but it's probably related (as it seems to be the only one not generated)

Reply:
I think this piece of code needs to be added to collectBases:
```
    case Defn.Object(_, _, Template(_, _, _, stats)) =>
      stats.foreach(collectBases(_))
```
@niomaster Should I create a PR for this? (Just making sure I'm not duplicating work)

Reply:
ah, that seems right. Perhaps also `collectFamily`?

Reply:
Sure, I'll create a PR.